File: apprefurbished/src/main/java/com/androidcollider/easyfin/accounts/add_edit/AddAccountModule.java
package com.androidcollider.easyfin.accounts.add_edit;

import android.content.Context;

import com.androidcollider.easyfin.common.managers.accounts_info.AccountsInfoManager;
import com.androidcollider.easyfin.common.managers.format.number.NumberFormatManager;
import com.androidcollider.easyfin.common.managers.resources.ResourcesManager;
import com.androidcollider.easyfin.common.repository.Repository;

import dagger.Module;
import dagger.Provides;

/**
 * @author Ihor Bilous
 */

@Module
public class AddAccountModule {

    @Provides
    AddAccountMVP.Model provideAddAccountMVPModel(Repository repository,
                                                  AccountsInfoManager accountsInfoManager,
                                                  NumberFormatManager numberFormatManager,
                                                  ResourcesManager resourcesManager) {
        return new AddAccountModel(repository, accountsInfoManager, numberFormatManager, resourcesManager);
    }

    @Provides
    AddAccountMVP.Presenter provideAddAccountMVPPresenter(AddAccountMVP.Model model,
                                                          Context context) {
        return new AddAccountPresenter(model, context);
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/accounts/add_edit/AddAccountModule.kt
package com.androidcollider.easyfin.accounts.add_edit

import android.content.Context
import com.androidcollider.easyfin.common.managers.accounts.accounts_info.AccountsInfoManager
import com.androidcollider.easyfin.common.managers.format.number.NumberFormatManager
import com.androidcollider.easyfin.common.managers.resources.ResourcesManager
import com.androidcollider.easyfin.common.repository.Repository
import dagger.Module
import dagger.Provides

/**
 * @author Ihor Bilous
 */
@Module
class AddAccountModule {
    @Provides
    fun provideAddAccountMVPModel(
        repository: Repository,
        accountsInfoManager: AccountsInfoManager,
        numberFormatManager: NumberFormatManager,
        resourcesManager: ResourcesManager
    ): AddAccountMVP.Model {
        return AddAccountModel(
            repository,
            accountsInfoManager,
            numberFormatManager,
            resourcesManager
        )
    }

    @Provides
    fun provideAddAccountMVPPresenter(
        model: AddAccountMVP.Model,
        context: Context
    ): AddAccountMVP.Presenter {
        return AddAccountPresenter(model, context)
    }
}